src/shared/utils.ts

```diff
@@ -233,13 +233,32 @@ export async function willForceUpload(node: IZoweDatasetTreeNode | IZoweUSSTreeN
                 title
             }, () => {
                 if (isZoweDatasetTreeNode(node)) {
-                    return ZoweExplorerApiRegister.getMvsApi(node ? node.getProfile(): profile).putContents(doc.fileName,
+                    const prof = node ? node.getProfile(): profile;
+                    if (prof.profile.encoding) {
+                        uploadOptions.encoding = prof.profile.encoding;
+                    }
+                    return ZoweExplorerApiRegister.getMvsApi(prof).putContents(doc.fileName,
                         remotePath,
                         uploadOptions);
                 } else {
-                    return ZoweExplorerApiRegister.getUssApi(profile).putContents(
-                        doc.fileName, remotePath, binary, null, null, returnEtag);
+
+                    // if new api method exists, use it
+                    if (ZoweExplorerApiRegister.getUssApi(profile).putContent) {
+                        return ZoweExplorerApiRegister.getUssApi(profile).putContent(
+                            doc.fileName, remotePath,
+                            {
+                                binary,
+                                localEncoding: null,
+                                etag: null,
+                                returnEtag,
+                                encoding: profile.profile.encoding
+                            });
+                    } else {
+                        return ZoweExplorerApiRegister.getUssApi(profile).putContents(
+                            doc.fileName, remotePath, binary, null, null, returnEtag);
                     }
```

Comment:
Going back to my previous comment, if we deprecate `putContents`, by constructing the object in it and calling `putContent` we could reduce this logic 😋

Reply:
Is that possible here?  If the new method exists, I use it to pass on encoding.  If the new method doesn't exist I call the old method without encoding.  Or am I missing it entirely?  😄